airflow/configuration.py

```diff
@@ -350,10 +350,11 @@ def _using_old_value(self, old, current_value):
         return old.search(current_value) is not None
 
     def _update_env_var(self, section, name, new_value):
-        # Make sure the env var option is removed, otherwise it
-        # would be read and used instead of the value we set
         env_var = self._env_var_name(section, name)
-        os.environ.pop(env_var, None)
+        # If the config comes from environment, set it there so that any subprocesses keep the same override!
+        if os.environ.get(env_var):
```

Comment:
```suggestion
        if env_var in os.environ:
```

This matters because it is possible to set an environment variable to an empty string.